src/mds/journal.cc

```diff
@@ -2849,6 +2849,27 @@ void dirfrag_rollback::decode(bufferlist::iterator &bl)
 // -----------------------
 // EExport
 
+void EExport::remove_blooms(CDir *dir)
+{
+  dir->remove_bloom();
+
+  std::list<CDir*> subdirs;
+  CDir::map_t::iterator it;
+  for (it = dir->begin(); it != dir->end(); ++it) {
+    CDentry *dn = it->second;
+    CInode *in = dn->get_linkage()->get_inode();
+
+    if (dn->get_linkage()->is_primary()) {
+      in->get_nested_dirfrags(subdirs);
```

Comment:
I must be misremembering how EExport works. Why do we need to go down one level to pull out the child CDir bloom filters, but not all the way to the leaves?